contracts/modules/licensing/PILicenseTemplate.sol

```diff
@@ -0,0 +1,402 @@
+        /* solhint-disable */
+        // NOTE: TOTAL_RNFT_SUPPLY = 1000 in trait with max_value. For numbers, don't add any display_type, so that
+        // they will show up in the "Ranking" section of the OpenSea UI.
+        return
+            string(
+                abi.encodePacked(
+                    '{"trait_type": "Commercial Use", "value": "',
+                    terms.commercialUse ? "true" : "false",
+                    '"},',
+                    '{"trait_type": "Commercial Attribution", "value": "',
+                    terms.commercialAttribution ? "true" : "false",
+                    '"},',
+                    '{"trait_type": "Commercial Revenue Share", "max_value": 1000, "value": ',
+                    terms.commercialRevShare.toString(),
+                    "},",
+                    '{"trait_type": "Commercial Revenue Celling", "value": ',
+                    terms.commercialRevCelling.toString(),
+                    "},",
+                    '{"trait_type": "Commercializer Check", "value": "',
+                    terms.commercializerChecker.toHexString(),
+                    // Skip on commercializerCheckerData as it's bytes as irrelevant for the user metadata
+                    '"},'
+                )
+            );
+        /* solhint-enable */
+    }
+
+    /// @dev Encodes the derivative traits of PIL policy into a JSON string for OpenSea
+    function _policyDerivativeTraitsToJson(PILTerms memory terms) internal pure returns (string memory) {
+        /* solhint-disable */
+        // NOTE: TOTAL_RNFT_SUPPLY = 1000 in trait with max_value. For numbers, don't add any display_type, so that
+        // they will show up in the "Ranking" section of the OpenSea UI.
+        return
+            string(
+                abi.encodePacked(
+                    '{"trait_type": "Derivatives Allowed", "value": "',
+                    terms.derivativesAllowed ? "true" : "false",
+                    '"},',
+                    '{"trait_type": "Derivatives Attribution", "value": "',
+                    terms.derivativesAttribution ? "true" : "false",
+                    '"},',
+                    '{"trait_type": "Derivatives Revenue Celling", "value": ',
+                    terms.derivativeRevCelling.toString(),
+                    "},",
+                    '{"trait_type": "Derivatives Approval", "value": "',
+                    terms.derivativesApproval ? "true" : "false",
+                    '"},',
+                    '{"trait_type": "Derivatives Reciprocal", "value": "',
+                    terms.derivativesReciprocal ? "true" : "false",
+                    '"},'
+                )
+            );
+        /* solhint-enable */
+    }
+
+    /// @dev Checks the configuration of commercial use and throws if the policy is not compliant
+    // solhint-disable-next-line code-complexity
+    function _verifyCommercialUse(PILTerms calldata terms) internal view {
+        if (!terms.commercialUse) {
```

Comment:
looks to me commercialUse is the root of the configuration for a list of terms, such like commercialAttribution, commercializerChecker, etc.  The current code seems complicated.  I understood it has to work with the current PIL.  We really shall look for optimization later.

Reply:
Will optimize in separate PR.